### DISCUSSION TITLE
[yocto, meta-flutter] libwebrtc

### DISCUSSION THREAD

**Discussion by jwinarske** [flutter-webrtc/flutter-webrtc#1701] [Show and tell]
I created a Yocto recipe for libwebrtc in meta-flutter (scarthgap):
https://github.com/meta-flutter/meta-flutter/blob/scarthgap/recipes-graphics/libwebrtc/libwebrtc_125.6422.06.bb

Support will be added to kirkstone and dunfell in the near future.

Also flutter-webrtc plugin support has been added to ivi-homescreen/flutter-auto:
https://github.com/toyota-connected/ivi-homescreen-plugins/tree/v2.0/plugins/webrtc

**Comment by zhuzhengjing** [comment]
| FAILED: obj/libwebrtc/libwebrtc/rtc_desktop_media_list_impl.o
| ../../third_party/llvm-build/Release+Asserts/bin/clang++ -MD -MF obj/libwebrtc/libwebrtc/rtc_desktop_media_list_impl.o.d -DUSE_LIBYUV -DWEBRTC_INCLUDE_INTERNAL_AUDIO_DEVICE -DWEBRTC_LINUX -DWEBRTC_POSIX -DRTC_DESKTOP_DEVICE -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_OZONE=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_GNU_SOURCE -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_EXTENSIVE -DCR_CLANG_REVISION=\"llvmorg-19-init-8091-gab037c4f-1\" -D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS -D_LIBCXXABI_DISABLE_VISIBILITY_ANNOTATIONS -DCR_LIBCXX_REVISION=e3b94d0e5b86883fd77696bf10dc33ba250ba99b -DCR_SYSROOT_KEY=20230611T210420Z-2 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DWEBRTC_ENABLE_PROTOBUF=0 -DWEBRTC_STRICT_FIELD_TRIALS=0 -DRTC_ENABLE_VP9 -DRTC_DAV1D_IN_INTERNAL_DECODER_FACTORY -DWEBRTC_HAVE_SCTP -DWEBRTC_USE_H264 -DWEBRTC_ENABLE_LIBEVENT -DWEBRTC_ARCH_ARM64 -DWEBRTC_HAS_NEON -DWEBRTC_ENABLE_AVX2 -DWEBRTC_NON_STATIC_TRACE_EVENT_HANDLERS=0 -DABSL_ALLOCATOR_NOTHROW=1 -DLIBYUV_DISABLE_LSX -DLIBYUV_DISABLE_LASX -DHAVE_WEBRTC_VIDEO -DMANGLE_JPEG_NAMES -I../../libwebrtc -I../../libwebrtc/include -I../.. -Igen -I../../buildtools/third_party/libc++ -I../../third_party/abseil-cpp -I../../third_party/libyuv/include -I../../third_party/boringssl/src/include -I../../third_party/libjpeg_turbo -Wall -Wextra -Wimplicit-fallthrough -Wextra-semi -Wunreachable-code-aggressive -Wthread-safety -Wno-missing-field-initializers -Wno-unused-parameter -Wno-psabi -Wloop-analysis -Wno-unneeded-internal-declaration -Wno-cast-function-type -Wno-ignored-pragma-optimize -Wno-deprecated-builtins -Wno-bitfield-constant-conversion -Wno-deprecated-this-capture -Wno-invalid-offsetof -Wno-vla-extension -Wno-thread-safety-reference-return -Wshadow -Werror -fno-delete-null-pointer-checks -fno-ident -fno-strict-aliasing -fstack-protector -funwind-tables -fPIC -pthread -fcolor-diagnostics -fmerge-all-constants -fcrash-diagnostics-dir=../../tools/clang/crashreports -mllvm -instcombine-lower-dbg-declare=0 -mllvm -split-threshold-for-reg-with-hint=0 -ffp-contract=off -fcomplete-member-pointers -mbranch-protection=standard --target=aarch64-linux-gnu -mno-outline -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -ffile-compilation-dir=. -no-canonical-prefixes -ftrivial-auto-var-init=pattern -O2 -fdata-sections -ffunction-sections -fno-unique-section-names -fno-math-errno -fno-omit-frame-pointer -g0 -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -Wno-shadow -Wctad-maybe-unsupported -Wc++11-narrowing -Wundef -Wunused-lambda-capture -isystem/home/annax/Documents/work/imx-yocto-bsp/build_imx8p-wayland/tmp/work/armv8a-poky-linux/libwebrtc/125.6422.06/recipe-sysroot/usr/include/gio-unix-2.0 -isystem/home/annax/Documents/work/imx-yocto-bsp/build_imx8p-wayland/tmp/work/armv8a-poky-linux/libwebrtc/125.6422.06/recipe-sysroot/usr/include/libmount -isystem/home/annax/Documents/work/imx-yocto-bsp/build_imx8p-wayland/tmp/work/armv8a-poky-linux/libwebrtc/125.6422.06/recipe-sysroot/usr/include/blkid -isystem/home/annax/Documents/work/imx-yocto-bsp/build_imx8p-wayland/tmp/work/armv8a-poky-linux/libwebrtc/125.6422.06/recipe-sysroot/usr/include/glib-2.0 -isystem/home/annax/Documents/work/imx-yocto-bsp/build_imx8p-wayland/tmp/work/armv8a-poky-linux/libwebrtc/125.6422.06/recipe-sysroot/usr/lib/glib-2.0/include -Wno-gcc-compat -Wno-unreachable-code-break -Wenum-compare-conditional -Wno-c++11-narrowing-const-reference -std=c++17 -Wno-trigraphs -gsimple-template-names -fno-exceptions -nostdinc++ -isystem../../third_party/libc++/src/include -isystem../../third_party/libc++abi/src/include --sysroot=../../../../recipe-sysroot -fvisibility-inlines-hidden -Wnon-virtual-dtor -Woverloaded-virtual -c ../../libwebrtc/src/rtc_desktop_media_list_impl.cc -o obj/libwebrtc/libwebrtc/rtc_desktop_media_list_impl.o
| ../../libwebrtc/src/rtc_desktop_media_list_impl.cc:46:14: error: no member named 'set_allow_pipewire' in 'webrtc::DesktopCaptureOptions'
|    46 |     options_.set_allow_pipewire(true);
|       |     ~~~~~~~~ ^
| 1 error generated.
